src/test/java/com/google/cloud/spanner/pgadapter/python/django/DjangoTransactionsTest.java

```diff
@@ -0,0 +1,166 @@
+
+    String actualOutput = executeTransactionTests(pgServer.getLocalPort(), host, options);
+    String expectedOutput = "Transaction Committed\n";
+
+    assertEquals(expectedOutput, actualOutput);
+    assertEquals(1, mockSpanner.countRequestsOfType(CommitRequest.class));
+
+  }
+
+  @Test
+  public void transactionRollbackTest() throws IOException, InterruptedException {
+
+    String updateSQL1 = "UPDATE \"singers\" SET \"firstname\" = 'hello', \"lastname\" = 'world' WHERE \"singers\".\"singerid\" = 1";
+    String insertSQL1 = "INSERT INTO \"singers\" (\"singerid\", \"firstname\", \"lastname\") VALUES (1, 'hello', 'world')";
+    mockSpanner.putStatementResult(StatementResult.update(Statement.of(updateSQL1), 0));
+    mockSpanner.putStatementResult(StatementResult.update(Statement.of(insertSQL1), 1));
+
+    String updateSQL2 = "UPDATE \"singers\" SET \"firstname\" = 'hello', \"lastname\" = 'python' WHERE \"singers\".\"singerid\" = 2";
+    String insertSQL2 = "INSERT INTO \"singers\" (\"singerid\", \"firstname\", \"lastname\") VALUES (2, 'hello', 'python')";
+    mockSpanner.putStatementResult(StatementResult.update(Statement.of(updateSQL2), 0));
+    mockSpanner.putStatementResult(StatementResult.update(Statement.of(insertSQL2), 1));
+
+    List<String> options = new ArrayList<String>();
+    options.add("rollback");
+
+    String actualOutput = executeTransactionTests(pgServer.getLocalPort(), host, options);
+    String expectedOutput = "Transaction Rollbacked\n";
+
+    assertEquals(expectedOutput, actualOutput);
+    assertEquals(1, mockSpanner.countRequestsOfType(RollbackRequest.class));
+
+  }
+
+  @Test
+  public void transactionAtomicTest() throws IOException, InterruptedException {
+
+    String updateSQL1 = "UPDATE \"singers\" SET \"firstname\" = 'hello', \"lastname\" = 'world' WHERE \"singers\".\"singerid\" = 1";
+    String insertSQL1 = "INSERT INTO \"singers\" (\"singerid\", \"firstname\", \"lastname\") VALUES (1, 'hello', 'world')";
+    mockSpanner.putStatementResult(StatementResult.update(Statement.of(updateSQL1), 0));
+    mockSpanner.putStatementResult(StatementResult.update(Statement.of(insertSQL1), 1));
+
+    String updateSQL2 = "UPDATE \"singers\" SET \"firstname\" = 'hello', \"lastname\" = 'python' WHERE \"singers\".\"singerid\" = 2";
+    String insertSQL2 = "INSERT INTO \"singers\" (\"singerid\", \"firstname\", \"lastname\") VALUES (2, 'hello', 'python')";
+    mockSpanner.putStatementResult(StatementResult.update(Statement.of(updateSQL2), 0));
+    mockSpanner.putStatementResult(StatementResult.update(Statement.of(insertSQL2), 1));
+
+    List<String> options = new ArrayList<String>();
+    options.add("atomic");
+
+    String actualOutput = executeTransactionTests(pgServer.getLocalPort(), host, options);
+    String expectedOutput = "Atomic Successful\n";
+
+    assertEquals(expectedOutput, actualOutput);
+    assertEquals(1, mockSpanner.countRequestsOfType(CommitRequest.class));
+
+  }
+
+
+
```

Comment:
nit: remove empty lines. It seems that you need to run the code formatter for this PR. This is done automatically when you run `mvn test`, or you can run only the code formatter with `mvn com.coveo:fmt-maven-plugin:format`.